vampyr/world/world.cpp

```diff
@@ -0,0 +1,67 @@
+#include <array>
+
+#include "pybind11/pybind11.h"
+#include "pybind11/stl.h"
+
+#include "trees/BoundingBox.h"
+#include "trees/MultiResolutionAnalysis.h"
+
+
+using namespace mrcpp;
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace vampyr {
+
+void world(py::module &m) {
+    const auto D = 3;
+    py::class_<BoundingBox<D>>(m, "BoundingBox")
+        .def(py::init<int, std::array<int, D>, std::array<int, D>>(),
+             "Initiates the BoundingBox of the area functions can exist")
+        .def(py::init<std::array<double, D>, bool>(),
+             "Initiates the BoundingBox of the area functions can exist, for periodic systems")
+        .def("getBoxLength", &BoundingBox<D>::getBoxLength)
+        .def("getBoxLengths", &BoundingBox<D>::getBoxLengths)
+        .def("getUpperBounds", &BoundingBox<D>::getUpperBounds)
+        .def("getUpperBounds", &BoundingBox<D>::getUpperBound)
+        .def("getLowerBounds", &BoundingBox<D>::getLowerBounds)
+        .def("getLowerBound", &BoundingBox<D>::getLowerBound)
+        .def("getUnitLength", &BoundingBox<D>::getUnitLength)
+        .def("getScalingFactor", py::overload_cast<int>(&BoundingBox<D>::getScalingFactor, py::const_))
+        .def("getCornerIndex", &BoundingBox<D>::getCornerIndex)
+        .def("size", py::overload_cast<>(&BoundingBox<D>::size, py::const_))
+        .def("size", py::overload_cast<int>(&BoundingBox<D>::size, py::const_))
+        .def("getScale", &BoundingBox<D>::getScale);
+
+    py::class_<MultiResolutionAnalysis<D>>(m, "MultiResolutionAnalysis")
+        .def(py::init<BoundingBox<D>, ScalingBasis, int>(), "BoundingBox"_a, "Basis Function"_a, "Scale"_a)
+        .def("getOrder", &MultiResolutionAnalysis<D>::getOrder, "Returns the order of the scaling basis")
+        .def("getWorldBox", &MultiResolutionAnalysis<D>::getWorldBox)
+        .def("getScalingBasis", &MultiResolutionAnalysis<D>::getScalingBasis)
+        .def("getMaxDepth", &MultiResolutionAnalysis<D>::getMaxDepth)
+        .def("getMaxScale", &MultiResolutionAnalysis<D>::getMaxScale)
+        .def("print", [](const MultiResolutionAnalysis<D> &a) {
```

Comment:
I would make more sense to have `__repr__` binding here, so that calling `print` in Python with an object of this type gives this string. This avoids mixing `py::print` in C++ code (which I am not very fond of)